### ISSUE TITLE
Failing pip3 install "faust[rocksdb,uvloop,fast,redis]" 

### ISSUE THREAD

**Issue by LodeVanB** [robinhood/faust#313]
## Checklist

- [ ] I have included information about relevant versions
- [ ] I have verified that the issue persists when using the `master` branch of Faust.

## Steps to reproduce

pip3 install "faust[rocksdb,uvloop,fast,redis]"

## Expected behavior 

no errors and certainly no missing Python.h (which is more than once found with command : "locate Python.h")

## Actual behavior

Tell us what happened instead.

.cache/pip/wheels/f2/9a/90/de94f8556265ddc9d9c8b271b0f63e57b26fb1d67a45564511
Successfully built colorclass terminaltables robinhood-aiokafka aredis idna-ssl contextvars pycparser
Failed to build setproctitle ciso8601 python-rocksdb
Installing collected packages: immutables, contextvars, aiocontextvars, colorlog, mypy-extensions, mode, click, six, python-dateutil, croniter, venusian, colorclass, multidict, idna, yarl, async-timeout, typing-extensions, attrs, chardet, idna-ssl, aiohttp, terminaltables, robinhood-kafka-python, robinhood-aiokafka, cchardet, typing, pycparser, cffi, pycares, aiodns, setproctitle, ciso8601, aredis, setuptools, python-rocksdb, uvloop, faust
  Running setup.py install for setproctitle ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-hjyinzdm/setproctitle/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-wh_1arzt-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_ext
    building 'setproctitle' extension
    creating build
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DSPT_VERSION=1.1.10 -DHAVE_SYS_PRCTL_H=1 -I/usr/include/python3.6m -c src/setproctitle.c -o build/temp.linux-x86_64-3.6/src/setproctitle.o
    In file included from src/spt.h:15:0,
                     from src/setproctitle.c:14:
    src/spt_python.h:14:10: fatal error: Python.h: No such file or directory
     #include <Python.h>
              ^~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-hjyinzdm/setproctitle/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-wh_1arzt-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-hjyinzdm/setproctitle/

.cache/pip/wheels/f2/9a/90/de94f8556265ddc9d9c8b271b0f63e57b26fb1d67a45564511
Successfully built colorclass terminaltables robinhood-aiokafka aredis idna-ssl contextvars pycparser
Failed to build setproctitle ciso8601 python-rocksdb
Installing collected packages: immutables, contextvars, aiocontextvars, colorlog, mypy-extensions, mode, click, six, python-dateutil, croniter, venusian, colorclass, multidict, idna, yarl, async-timeout, typing-extensions, attrs, chardet, idna-ssl, aiohttp, terminaltables, robinhood-kafka-python, robinhood-aiokafka, cchardet, typing, pycparser, cffi, pycares, aiodns, setproctitle, ciso8601, aredis, setuptools, python-rocksdb, uvloop, faust
  Running setup.py install for setproctitle ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-hjyinzdm/setproctitle/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-wh_1arzt-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_ext
    building 'setproctitle' extension
    creating build
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DSPT_VERSION=1.1.10 -DHAVE_SYS_PRCTL_H=1 -I/usr/include/python3.6m -c src/setproctitle.c -o build/temp.linux-x86_64-3.6/src/setproctitle.o
    In file included from src/spt.h:15:0,
                     from src/setproctitle.c:14:
    src/spt_python.h:14:10: fatal error: Python.h: No such file or directory
     #include <Python.h>
              ^~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-hjyinzdm/setproctitle/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-wh_1arzt-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-hjyinzdm/setproctitle/

# Versions

* Python3.6
* Faust version
* Operating system
* Kafka version
* RocksDB version (if applicable)

**Comment by ask** [comment]
Thank you for the report!

The ``fast`` option requires having a working C compiler and the Python header files.

If you're using Ubuntu this is solved by installing the Python header package:

```sudo apt-get install python3-dev```

You also need gcc and so on (build-essential package).

The setup we use to build Python packages on Ubuntu is:

```sudo apt-get install build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev```

**Comment by LodeVanB** [comment]
I already have those on my system, and nevertheless it doesn't find the header files! Quid?

build-essential is already the newest version (12.4ubuntu1).
libbz2-dev is already the newest version (1.0.6-8.1).
libreadline-dev is already the newest version (7.0-3).
libsqlite3-dev is already the newest version (3.22.0-1).
zlib1g-dev is already the newest version (1:1.2.11.dfsg-0ubuntu2).
libssl-dev is already the newest version (1.1.0g-2ubuntu4.3).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
python3-dev is already the newest version (3.6.7-1~18.04).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

**Comment by ask** [comment]
Hmm. Is pip3 using the system python?

**Comment by ask** [comment]
Not sure if you need python-dev, but that’s probabøy Python 2 only

**Comment by LodeVanB** [comment]
I have both P2 and P3 but use 3 most ... I test Python on various distro ... this one now is Mint Cinnamon 19 (Other Linux distro are Ubuntu LTS, Manjaro, Fedora 29, MX Linux, Alien OS, Kali Linux and Knoppix)

**Comment by osamaimran** [comment]
Yeah I have the same issue and all things I have installed as @LodeVanB 
anyone resolved this??
